Lib/datetime.py

```diff
@@ -1351,7 +1418,7 @@ def _cmp(self, other, allow_mixed=False):
                          other._microsecond))
```

Comment:
Why are comparisons ignoring the new field? Are two times that differ only by nanoseconds supposed to be equal?